### DISCUSSION TITLE
Phantom creation

### DISCUSSION THREAD

**Discussion by katyabrui** [JuliaHealth/KomaMRI.jl#312] [Q&A] [answered]
Hello! First of all, thank you for a great work, the simulator is so easy to use and fast!
I have small difficulties in creation and using a new phantom.
I have tried to make a 3D brain phantom using Phantom.jl and succeeded to make an obj struct. However, now I stuck with how to use it as a phantom in simulations from GUI.

**Comment by cncastillo** [comment]
Hi! perhaps this should be included in the docs in "Creating your own phantom" (@beorostica could add this? :pray:), but you can directly modify the struct that is being used be the GUI by doing the following in the REPL
```julia
obj_ui[] = your_phantom
```
This is further described in https://juliahealth.org/KomaMRI.jl/stable/ui-details/#REPL-and-UI-communication. Ideally, we would want to write the phantom and load it in the GUI, but that is currently being worked on in #184, so for the moment this should work :smiley:.

Please do not hesitate to ask if you have any further questions!

**Reply by katyabrui** [reply]
Thank you for the fast unswer!
I'm not familiar with Julia, unfortunatelly, so I've tried several variants of modifying the obj_ui but it hasn't work.
I used this function to create a phantom : obj = brain_phantom3D(),  then launched KomaUI, tried to change the phantom by 
obj_ui[ ] = obj,
but got an error message "`obj_ui` not defined"
 then I tried 
obj_ui[ ] = obj(), and got "MethodError: objects of type Phantom{Float64} are not callable"

**Reply by cncastillo** [reply, marked as answer]
What version of Koma are you using? (It should tell you the version when launching the GUI). If it is not 0.8, try updating first by typing `] up` in the Julia REPL. You should see something like this 
![image](https://github.com/JuliaHealth/KomaMRI.jl/assets/5957134/ac8a4251-cd35-4029-a16a-7590c07a3e61)

**Reply by katyabrui** [reply]
I have no Cuda capable devices for 0.8.0 on my computer=)) But on the other computer it works, thank you very much!
The next step will be dealing with 3D simulations=))

**Reply by cncastillo** [reply]
Are you also doing 3D sequences? The image reconstruction in the UI could have some problems, so let me know if you encounter any issues. Any other doubt just ask! :)

**Reply by katyabrui** [reply]
Yes, we tried MPRAGE and it was not reconstructed propperly. So we just wrote a script to reconstruct it =)

**Reply by cncastillo** [reply]
Could you provide a version of the MPRAGE sequence so we can make this work? or even better, create an issue about the recon for 3d sequences not working :pray:  Did you export the raw data from the UI for the recon? Just curious. 

Thanks for the feedback :smile:

**Reply by katyabrui** [reply]
Yes, we used the raw data from Koma. About MPRAGE, yes, of course. I'm not sure that our MPRAGE is already ideal ... but anyway I will attach a .seq file to a new issue.
